### PR TITLE
Terminal Abelian category

### PR DESCRIPTION
**Pull request by felixwellen**
Construct the terminal abelian category.
This builds on unmerged work of @barrettj12 and should be merged after #672 .

### CODE REVIEW

File: Cubical/Algebra/Group/Instances/Unit.agda [hunk 1]
```diff
@@ -25,6 +25,12 @@ Unit = UnitType , groupstr tt (λ _ _ → tt) (λ _ → tt)
                       (makeIsGroup isSetUnit (λ _ _ _ → refl) (λ _ → refl) (λ _ → refl)
                                    (λ _ → refl) (λ _ → refl))
 
+Unit* : Group ℓ
```
Comment (mortberg) [inline Cubical/Algebra/Group/Instances/Unit.agda:28]: UnitGroup

write with copatterns

File: Cubical/Categories/Abelian/Instances/Terminal.agda [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+private
+  variable
+    ℓ : Level
+
+private
+  open PreAbCategory
+  open PreAbCategoryStr
+
+  terminalPreAb : PreAbCategory ℓ ℓ
```
Comment (mortberg) [inline Cubical/Categories/Abelian/Instances/Terminal.agda:19]: terminalPreAbCategory ?

File: Cubical/Categories/Additive/Instances/Terminal.agda [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Categories.Category
+open import Cubical.Categories.Additive.Base
+open import Cubical.Categories.Instances.Discrete
+
+open import Cubical.Algebra.AbGroup
+open import Cubical.Algebra.AbGroup.Instances.Unit renaming (Unit* to UnitAbGroup)
```
Comment (mortberg) [inline Cubical/Categories/Additive/Instances/Terminal.agda:13]: Lol

File: Cubical/Categories/Additive/Instances/Terminal.agda [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  variable
+    ℓ ℓ' : Level
+
+open PreaddCategory
+open PreaddCategoryStr
+open AdditiveCategory
+open AdditiveCategoryStr
+
+private
+  terminalCat : Category ℓ ℓ
```
Comment (mortberg) [inline Cubical/Categories/Additive/Instances/Terminal.agda:27]: terminalCategory ?

File: Cubical/Categories/Additive/Instances/Terminal.agda [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  homAbStr (preadd terminalPreAdd) =
+    λ x y →
+      subst
+        AbGroupStr
+        (sym (isContr→≡Unit* (homContr x y)))
+        terminalAbGroupStr
+  ⋆distl+ (preadd terminalPreAdd) = λ _ _ _ _ → refl
+  ⋆distr+ (preadd terminalPreAdd) = λ _ _ _ → refl
+
+terminal : AdditiveCategory ℓ ℓ
```
Comment (mortberg) [inline Cubical/Categories/Additive/Instances/Terminal.agda:50]: terminalAdditiveCategory

File: Cubical/Categories/Additive/Instances/Terminal.agda [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  terminalAbGroupStr : AbGroupStr {ℓ = ℓ} Unit*
+  terminalAbGroupStr = snd UnitAbGroup
+
+  homProp : (x y : Category.ob {ℓ = ℓ} terminalCat) → isProp (terminalCat [ x , y ])
+  homProp x y = isOfHLevelUnit* 2 x y
+
+  homContr : (x y : Category.ob {ℓ = ℓ} terminalCat) → isContr (terminalCat [ x , y ])
+  homContr x y = isProp→isContrPath (isOfHLevelUnit* 1) x y
+
+  terminalPreAdd : PreaddCategory ℓ ℓ
```
Comment (mortberg) [inline Cubical/Categories/Additive/Instances/Terminal.agda:39]: terminalPreaddCategory

**Review by mortberg — APPROVED** [review]

File: Cubical/Categories/Additive/Instances/Terminal.agda (hunk 4 above)
Comment (felixwellen) [reply Cubical/Categories/Additive/Instances/Terminal.agda:27]: l' would need a different construction...

**Pull request merged**